app/views/posts/index.html.erb

```diff
@@ -1,2 +1,41 @@
-<h1>User's Posts Goes Here</h1>
-<p>Find every post of your favorite author here</p>
+<div class="w-full flex flex-col items-center">
+  <div class="bg-white p-4 rounded-lg w-4/5 shadow-md mb-8">
+    <div class="flex justify-between mb-4">
+      <div class="mr-4">
+        <% if @user.photo.present? %>
+          <%= image_tag @user.photo, class: 'w-16 h-16 rounded-full', alt: "#{@user.name}" %>
+        <% else %>
+          <%= image_tag 'placeholder.jpg', class: 'w-16 h-16 rounded-full', alt: 'Placeholder Image' %>
+        <% end %>
+      </div>
+
+      <div class="justify-evenly w-4/5">
+        <h3 class="text-lg text-start px-2 font-semibold mb-2 pr-2"><%= @user.name %></h3>
+        <p class="text-gray-600 text-end pr-2 ">Number of posts: <%= @user.posts.count %></p>
+      </div>
+    </div>
+  </div>
+
+  <% @posts.each do |post| %>
+    <div class="bg-white p-4 w-4/5 rounded-lg shadow-md mb-4">
+      <h4 class="text-lg mb-2"><%= link_to "Post #{post.id}", user_post_path(@user, post) %> </h4>
+      <p class="text-gray-800"><%= post.text %></p>
+      
+      <div class="flex justify-end mt-4">
+        <p class="text-gray-600  pr-2">Comments: <%= post.comments.count %></p>
+        <p class="text-gray-600">Likes: <%= post.likes.count %></p>
+      </div>
+
+      <ul>
+        <% post.comments.limit(5).each do |comment| %>
+          <li>
+            <strong><%= comment.user.name %></strong>: <%= comment.text %>
+          </li>
+        <% end %>
+      </ul>
```

Comment:
-[ ] Instead of hardcoding the 5 recent comments within the `post#index` view, let follow best practices by using the `recent_comment` method created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md) to render the 5 most recent comments, similar to the code snippet below:

```ruby
# app/views/posts/index.html.erb

 <% post.recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>
```

Reply:
Thanks for the constructive suggestion; I have implemented now